style-guide.md

```diff
@@ -0,0 +1,112 @@
+
+## Accessibility
+See also: https://developers.google.com/style/accessibility
+
+* Always set descriptive alt text for images, except for decorative images
+ 	* If the image contains important text, repeat that text in the alt text
+	 	* Although it is tricky due to how documentation websites tend to use images to clarify written text,  try to avoid having alt text repeat what is in the body text
+	 	* Do not start with "image of" or "picture of" but do describe the type of image, like:
+	 		* "Headshot of David Haussler smiling"
+		 	* "Screenshot of the UI of dockstore.org/my-tools with the Refresh Organization button circled"
+		 	* "Animation of a tabby cat jumping three feet into the air upon noticing a cucumber"
+* Keep in mind ~5% of the population is red-green colorblind
+	* 	When color schemes convey non-essential meaning, avoid using color schemes that would be inaccessible to red-green colorblind users
+	* 	When color schemes convey essential meaning (vital UI elements, etc) they should be functional in grayscale
+* Do not use things that would be a nightmare for screen readers
+ 	* Avoid using non-ASCII symbols unless you are writing something meaningful in another language
+ 		* For example, do not use symbols from U+1D400 𝒔𝒖𝒄𝒉 𝒂𝒔 𝒕𝒉𝒆𝒔𝒆 as a replacement for ***actual bold and italic formatting***
+		* Using actual bold and italics formatting is acceptable, but bear in mind screen readers may not declare something as being bold or italic
+
+## Assumptions about a user's system
+* If talking about a program that does not support certain operating systems, make it clear which systems are supported relatively early in documentation introducing that program
+	* If possible, give an alternative for unsupported OSs
+	* "The Dockstore CLI is only supported on Unix-like operating systems, including Linux and Mac. Windows users may wish to..."
+* Do not assume users are or are not using an HPC
+	* Assume that if they are using an HPC, they at least know how to use their HPC's job submission system (SLURM, etc)
+	* If working with/installing something requiring root permissions such as Docker, mention that root permission is needed once, then assume from that point forward that your HPC users have root privileges
+	* However, do not assume that HPC users know how programs you introduce may work differently on an HPC
+		* Example: Running a WDL on an HPC via Cromwell (ie via the Dockstore CLI) requires additional runtime arguments that a local user does not need to include in their WDL program, so make sure to either include those runtime arguments in your examples or tell HPC users what additional things they need
+* Assume Linux users are using Ubuntu or know how to translate into their own distro
+	* Exception: If directing very new users to install software via apt-get, consider mentioning that the instructions are specific to Ubuntu -- UCSC's linux lab for students runs on CentOS (which uses yum instead) so this is not a rare occurrence
```

Comment:
Why assume Ubuntu?  (possibly a completely-valid assumption but I'm curious about the evidence)

Reply:
The dockstore quickstart for installing the Dockstore CLI explicitly says "Linux/Ubuntu" and calls apt-get, so I'm following its example.